### ISSUE TITLE
Campain Emails with embeded pictures

### ISSUE THREAD

**Issue by Lehnerr** [salesagility/SuiteCRM#5662]
#### Issue

When I create Emailtemplates with Mozaik-Editor and load the images to the system. The pictures are not displayed in the sent email.
![grafik](https://user-images.githubusercontent.com/16813374/38362976-15091b4e-38d3-11e8-8525-ff8e2dcc04f3.png)

#### Expected Behavior

The Pictures should be displayed.
#### Actual Behavior

Pictures are not displayed
#### Possible Fix

The problem is in the parsing of the Template. modules\EmailTemplates\EmailTemplate.php
    public function addDomainToRelativeImagesSrc() {
        global $sugar_config;
        $domain = $sugar_config['site_url'] . '/';
        $ret = $this->body_html = preg_replace('/(&lt;img src=&quot;)(public\/[^.]*.(jpg|jpeg|png|gif|bmp))(&quot;)/', "$1" . $domain . "$2$4", $this->body_html);
        return $ret;
    }

This is looking for &lt;img src=&quot;  But in the template  it is stored in following format:
f; font-size: 14px;&quot; alt=&quot;&quot; src=&quot;public/d09e3294-aba0-ed22-a101-5ac37a64eada.jpg&quot; data-mce-s

I changed th function to following and now it works:
    public function addDomainToRelativeImagesSrc() {
        global $sugar_config;
        $domain = $sugar_config['site_url'] . '/';
        $ret = $this->body_html = preg_replace('/(&lt;img src=&quot;)(public\/[^.]*.(jpg|jpeg|png|gif|bmp))(&quot;)/', "$1" . $domain . "$2$4", $this->body_html);
        $ret = $this->body_html = preg_replace('/( src=&quot;)(public\/[^.]*.(jpg|jpeg|png|gif|bmp))(&quot;)/', "$1" . $domain . "$2$4", $ret);
        return $ret;
    }

Pls fix it in Source

#### Steps to Reproduce

1. Create Campaign
2.In the EmailTemplate  upload a png file 
3.Send the Test-Email

#### Your Environment

* SuiteCRM Version used: 7.8.18, 7.10.2
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): happens in all browsers
* Environment name and version (e.g. MySQL, PHP 7): 7.0
* Operating System and version (e.g Ubuntu 16.04): Ubunto 16.04

**Comment by ebogaard** [comment]
I can confirm this is still happening in 7.8.17.
We fixed it by changing the preg_replace to:

`         $ret = $this->body_html = preg_replace('/( src=&quot)(public\/.*(jpg|jpeg|png|gif|bmp))(&quot)/', "$1" . $domain . "$2$4", $this->body_html);`

Quotes are html encoded in the source, so using an escaped quote in the regex won't do anything.

We found another strange 'functionality' with EmailTemplates: when adding/uploading images this way, the image is attached to the template as well. As in the EmailTemplate the image is referred to with it's URL, it seems to me attaching the image to the mail is a bit redundant.

**Comment by pstevens71** [comment]
As a work around until this is fixed, what I've been doing is just adding a full url in the image source (it has to be different from the installation, or it will remove the site address and make it relational).  So if your site is: https://mysite.com/public/...etc,     edit the image in your editor and add http://mysite.com/  before the relational URL  (notice its http, otherwise on save it will remove it and make it relational again).  That way the images work with the full URL.

**Comment by pstevens71** [comment]
I have a discussion going about this issue in the SuiteCRM forum.... here's what I've learned so far:

1. The proposed fixes on Github don't work. (I tried them both).
2. It's not a security issue or a site URL issue, I know this because .PNG files work. (and my permissions and site url are good).
3. If you upload a .PNG file works no problem and sends email with proper embedded image.
4. If you usea .JPG it shows in editor, but when you save it disappears and also does not send it in the email.
5. .GIF files will not upload at all.

**Comment by lazka** [comment]
I've opened a PR for this: #7046

**Comment by jfd118** [comment]
Hello SuiteCRM Developers, 

According to the release notes of 7.11.3, the problem this thread discusses was fixed.
github.com/salesagility/SuiteCRM/issues/5662

I just upgraded to 7.11.3 and conducted a test that seems to have failed. My process was:

1) Create a new email template using the Moziak email editor 
2) Insert an image to into the template. Save the template.  NOTE: when I upload the image I can resize the image amongst the text and the image appears while resizing. 
3) I selected my email address from the "targets" list and the email editor pops up.
4) I select the template I just created containing the image. The template has a place holder for the image but image is not there.
5) Send the email to myself.
6) Receive email. Only a placeholder for the image is in the message. The image itself is not.

First off, my set up is:
Debian 9
Package contents
Apache 2.4.33
MySQL 5.7.22
OpenSSL 1.0.2o
PHP 7.0.30
phpMyAdmin 4.8.1
SQLite 3.18.0
**SuiteCRM 7.11.3**

Thanks for your consideration, 

Jan
